src/editor.rs

```diff
@@ -105,21 +106,24 @@ macro_rules! cur_buf {
     }
 }
 
-impl<'a, W: Write> Editor<'a, W> {
-    pub fn new<P: Into<String>>(out: W, prompt: P, context: &'a mut Context) -> io::Result<Self> {
-        Editor::new_with_init_buffer(out, prompt, context, Buffer::new())
+impl<'a, 'b, W: Write> Editor<'a, W> {
+    pub fn new<P: Into<String>, F: 'static>(out: W, prompt: P, f: F, context: &'a mut Context) -> io::Result<Self>
+    where for<'r> F: (Fn(&'r str,) -> String) {
+        Editor::new_with_init_buffer(out, prompt, Box::new(f), context, Buffer::new())
     }
 
     pub fn new_with_init_buffer<P: Into<String>, B: Into<Buffer>>(
         out: W,
         prompt: P,
+        f: Box<Fn(&str) -> String>,
```

Comment:
Here you're taking a box, but in `new` you're taking a generic parameter. Perhaps this could also be generic, so it doesn't stand out?

Reply:
Done.